src/game_inventory.cpp

```diff
@@ -625,11 +631,69 @@ class comestible_inventory_preset : public inventory_selector_preset
 
 item_location game_menus::inv::consume( player &p )
 {
+    if( !g->u.has_activity( activity_id( "ACT_EAT_MENU" ) ) ) {
+        g->u.assign_activity( activity_id( "ACT_EAT_MENU" ) );
+    }
+
     return inv_internal( p, comestible_inventory_preset( p ),
                          _( "Consume item" ), 1,
                          _( "You have nothing to consume." ) );
 }
 
+class comestible_filtered_inventory_preset : public comestible_inventory_preset
+{
+    public:
+        comestible_filtered_inventory_preset( const player &p, bool( *predicate )( item it ) ) :
```

Comment:
```suggestion
        comestible_filtered_inventory_preset( const player &p, bool( *predicate )( const item &it ) ) :
```

Reply:
This and the other suggested changes from `item it` to `const item &it` are to ensure items are passed by reference, rather than creating new items.